core/controlplane/config/config.go

```diff
@@ -345,15 +317,17 @@ func (c *Cluster) SetDefaults() {
 	}
 
 	if len(c.Controller.Subnets) == 0 {
-		if privateTopologyImplied {
-			c.Controller.Subnets = c.PrivateSubnets()
-		} else {
-			c.Controller.Subnets = c.PublicSubnets()
+		c.Controller.Subnets = c.PublicSubnets()
```

Comment:
I was always lost in this part of code. Why `PublicSubnets` is better default than `PrivateSubnets()`?

Reply:
That's because you have no private subnets by default.
Anyway, I guess we can discuss it for
alternatively defaulting to either PrivateSubnets or just Subnets in a separate issue!